test/unit/data/dem_data.test.js

```diff
@@ -0,0 +1,213 @@
+        t.end();
+    });
+
+    t.test('constructor defaults', (t) => {
+        const dem = new DEMData(0);
+        t.false(dem.loaded);
+        t.equal(dem.uid, 0);
+        t.equal(dem.scale, 1);
+
+        t.end();
+    });
+
+    t.test('constructor with data', (t) => {
+        const data = new Level(16, 8, new Int32Array(32 * 32));
+        const dem = new DEMData(0, 1, data);
+        t.equal(dem.level.dim, 16);
+        t.equal(dem.level.border, 8);
+
+        t.end();
+    });
+
+    t.end();
+});
+
+
+test('DEMData#backfillBorder', (t) => {
+    const imageData0 = createMockImage(4, 4);
+    const dem0 = new DEMData(0);
+    const imageData1 = createMockImage(4, 4);
+    const dem1 = new DEMData(1);
+
+    dem0.loadFromImage(imageData0);
+    dem1.loadFromImage(imageData1);
+
+    t.test('border region is initially populated with neighboring data', (t)=>{
+        const level0 = dem0.level;
+        let nonempty = true;
+        for (let x = -1; x < 5; x++) {
+            for (let y = -1; y < 5; y++) {
+                if (level0.get(x, y) === -65536) {
+                    nonempty = false;
+                    break;
+                }
+            }
+        }
+        t.true(nonempty, 'pixel data populates DEM data level');
+
+        let verticalBorderMatch = true;
+        for (let x = -1; x < 5; x += 5) {
+            for (let y = 0; y < 4; y++) {
+                if (level0.get(x, y) !== level0.get(x < 0 ? x + 1 : x - 1, y)) {
+                    verticalBorderMatch = false;
+                    break;
+                }
+            }
+        }
+        t.true(verticalBorderMatch, 'vertical border of DEM data is initially equal to next column of data');
+
+        // horizontal borders empty
+        let horizontalBorderMatch = true;
+        for (let y = -1; y < 5; y += 5) {
+            for (let x = 0; x < 4; x++) {
+                if (level0.get(x, y) !== level0.get(x, y < 0 ? y + 1 : y - 1)) {
+                    horizontalBorderMatch = false;
+                    break;
+                }
+            }
+        }
+        t.true(horizontalBorderMatch, 'horizontal border of DEM data is initially equal to next row of data');
+
+        t.true(level0.get(-1, 4) === level0.get(0, 3), '-1, 1 corner initially equal to closest corner data');
+        t.true(level0.get(4, 4) === level0.get(3, 3), '1, 1 corner initially equal to closest corner data');
+        t.true(level0.get(-1, -1) === level0.get(0, 0), '-1, -1 corner initially equal to closest corner data');
+        t.true(level0.get(4, -1) === level0.get(3, 0), '-1, 1 corner initially equal to closest corner data');
+
+
+        t.end();
+    });
+
+    t.test('backfillBorder correctly populates borders with neighboring data', (t)=>{
+        const level0 = dem0.level;
+
+        dem0.backfillBorder(dem1, -1, 0);
+        for (let y = 0; y < 4; y++) {
+            t.true(level0.get(-1, y) !== level0.get(0, y), 'backfills neighbor -1, 0');
```

Comment:
This looks like it's checking that the border no longer matches the closest pixel from the content, but it doesn't check whether the new backfilled data has correct values.  Is there a good way to check the latter?

Reply:
😛 good point! yes I think it would be p simple to add this.